dev/benchmarks/macrobenchmarks/lib/src/web/recorder.dart

```diff
@@ -748,11 +806,36 @@ class AnnotatedSample {
 
 /// Base class for a profile collected from running a benchmark.
 class Profile {
-  Profile({@required this.name}) : assert(name != null);
+  Profile({@required this.name, this.useCustomWarmUp = false})
+      : assert(name != null),
+        _isWarmingUp = useCustomWarmUp;
 
   /// The name of the benchmark that produced this profile.
   final String name;
 
+  /// Whether to delimit warm-up frames in a custom way.
+  final bool useCustomWarmUp;
+
+  /// Whether we are measuring warm-up frames currently.
+  bool get isWarmingUp => _isWarmingUp;
+
+  bool _isWarmingUp;
+
+  /// Stop the warm-up phase.
+  ///
+  /// Call this method only when [useCustomWarmUp] and [isWarmingUp] are both
+  /// true.
+  /// Call this method only once for each profile.
+  void stopWarmingUp() {
+    if (! useCustomWarmUp) {
```

Comment:
no space between `!` and `useCustomWarmUp`

Reply:
Done.